iroh-bytes/src/store/redb/validate.rs

```diff
@@ -0,0 +1,362 @@
+                            let entry = v.value();
+                            trace!("blob {} -> {:?}", hash.to_hex(), entry);
+                        }
+                        Err(cause) => {
+                            error!("failed to access blob item: {}", cause);
+                        }
+                    }
+                }
+            }
+            Err(cause) => {
+                error!("failed to iterate blobs: {}", cause);
+            }
+        }
+        trace!("dumping inline_data");
+        match inline_data.iter() {
+            Ok(iter) => {
+                for item in iter {
+                    match item {
+                        Ok((k, v)) => {
+                            let hash = k.value();
+                            let data = v.value();
+                            trace!("inline_data {} -> {:?}", hash.to_hex(), data.len());
+                        }
+                        Err(cause) => {
+                            error!("failed to access inline data item: {}", cause);
+                        }
+                    }
+                }
+            }
+            Err(cause) => {
+                error!("failed to iterate inline_data: {}", cause);
+            }
+        }
+        trace!("dumping inline_outboard");
+        match inline_outboard.iter() {
+            Ok(iter) => {
+                for item in iter {
+                    match item {
+                        Ok((k, v)) => {
+                            let hash = k.value();
+                            let data = v.value();
+                            trace!("inline_outboard {} -> {:?}", hash.to_hex(), data.len());
+                        }
+                        Err(cause) => {
+                            error!("failed to access inline outboard item: {}", cause);
+                        }
+                    }
+                }
+            }
+            Err(cause) => {
+                error!("failed to iterate inline_outboard: {}", cause);
+            }
+        }
+        trace!("dumping tags");
+        match tags.iter() {
+            Ok(iter) => {
+                for item in iter {
+                    match item {
+                        Ok((k, v)) => {
+                            let tag = k.value();
+                            let value = v.value();
+                            trace!("tags {} -> {:?}", tag, value);
+                        }
+                        Err(cause) => {
+                            error!("failed to access tag item: {}", cause);
+                        }
+                    }
+                }
+            }
+            Err(cause) => {
+                error!("failed to iterate tags: {}", cause);
+            }
+        }
+
+        // perform consistency check for each entry
+        info!("validating blobs");
+        // set of a all hashes that are referenced by the blobs table
+        let mut entries = BTreeSet::new();
```

Comment:
this could become quite large, so it would be good to at least warn about this in the docs, that this can eat a lot of memory depending on the size of your store